javaslang/src/main/java/javaslang/collection/Array.java

```diff
@@ -899,38 +898,12 @@ public int length() {
 
     @Override
     public Array<T> removeAll(T element) {
-        final java.util.List<T> list = new ArrayList<>();
-        for (int i = 0; i < length(); i++) {
```

Comment:
as a bonus, this always results in the same `empty()` instance (if that's the case), with the new implementation